### DISCUSSION TITLE
Cloud integration for stantargets?

### DISCUSSION THREAD

**Discussion by mbjoseph** [wlandau/stantargets#37] [Help] [answered]
Thanks for the stantargets + targets packages - I've found them incredibly useful. Is there stantargets cloud integration similar to targets? 

The description of `resources` in the help file for `tar_stan_mcmc` suggests that AWS cloud storage may work, but I'm having a hard time pushing Stan targets to an S3 bucket, e.g.: 

```r
targets::tar_dir({ # tar_dir() runs code from a temporary directory.
  targets::tar_script({
    library(stantargets)
    # Do not use temporary storage for stan files in real projects
    # or else your targets will always rerun.
    path <- tempfile(pattern = "", fileext = ".stan")
    tar_stan_example_file(path = path)
    list(
      tar_stan_mcmc(
        your_model,
        stan_files = path,
        data = tar_stan_example_data(),
        variables = "beta",
        summaries = list(~quantile(.x, probs = c(0.25, 0.75))),
        stdout = R.utils::nullfile(),
        stderr = R.utils::nullfile(), 
        resources = list(bucket = "some-bucket-name")
      )
    )
  }, ask = FALSE)
  targets::tar_make()
})
```

This runs, but I don't see any targets populating the bucket `some-bucket-name`. Is this intended behavior? It would be great to be able to push to an S3 bucket for long-running models!

**Comment by wlandau** [comment, marked as answer]
For ordinary AWS-backed targets, users need to supply one of the `"aws_""` formats to the `format` argument of `tar_target()`. I chose to hide the `format`s in `tar_stan_mcmc()` because I thought there were clear choices for local storage and I was not thinking about the cloud. So this should probably be a feature.

**Reply by mbjoseph** [reply]
Great - thanks @wlandau!
